Unable to build outside of git checkout

Building outside of a git checkout fails with

```
fatal: not a git repository (or any of the parent directories): .git
CMake Error at cmake/GitRevision.cmake:4 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 128"

Call Stack (most recent call first):
  src/CMakeLists.txt:4 (git_rev_parse)
```

--- a/cmake/GitRevision.cmake
+++ b/cmake/GitRevision.cmake
@@ -1,12 +1,20 @@
 function(git_rev_parse output_var source_dir)
 
     if (NOT DEFINED ${output_var})
+        set(command git -C ${source_dir} rev-parse HEAD)
+
         execute_process(
-            COMMAND git -C ${source_dir} rev-parse HEAD
+            COMMAND ${command}
             OUTPUT_VARIABLE ${output_var}
             OUTPUT_STRIP_TRAILING_WHITESPACE
-            COMMAND_ERROR_IS_FATAL ANY
+            RESULT_VARIABLE result
         )
+
+        if (NOT result EQUAL 0)
+            list(JOIN command " " command)
+            message(WARNING "Command \"${command}\" failed; VCS information will be unavailable")
+            set (${output_var} "UNKNOWN")
+        endif()
     endif()
 
     message(STATUS "Got ${output_var}: ${${output_var}}")

